lib/ftp.c

```diff
@@ -1940,6 +1940,15 @@ static CURLcode ftp_state_pasv_resp(struct connectdata *conn,
 #endif
   {
     /* normal, direct, ftp connection */
+
+    /* postponed address resolution in case of tcp fastopen */
+    if (conn->bits.tcp_fastopen && !conn->bits.reuse && strlen(ftpc->newhost) == 0) {
```

Comment:
No need to get the length if you only want to check if if the first byte is zero. Saves cpu.

But perhaps more importantly: is this really good enough? I spot `ftpc->newhost` get assigned on three different places, and two of them strdup() the return value from `control_address()` ... which itself uses `conn->ip_addr_str`, so this update needs to be done *before* that doesn't it?

Reply:
My thoughts were, we only fill this field if we need it for the purpose of this PR:

`control_address()` derives its return value from either `conn->ip_addr_str` or ` conn->host_name`, depending on `conn->bits.tunnel_proxy || conn->bits.socksproxy`.
In the case of a a proxy is used, there is no need for `conn->ip_addr_str`, at least not for the purpose of this PR.
Also if `ftpc->newhost` has been set by other means, `ip_addr_str` is not required.
I only want to fill in `conn->ip_addr_str`, if it is the only means to get the remote host name into `fptc->newhost`.